app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/ElevationMapTest.java

```diff
@@ -0,0 +1,79 @@
+package com.github.ravifrancesco.softwaredevproject;
+
+import android.content.Context;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ElevationMapTest {
+/**
+ * Test geoTIFF API helper class constructor
+ */
+    @Test
+    public void geoTIFFConstructorTest() {
+
+        Context mContext = ApplicationProvider.getApplicationContext();
+
+        UserPoint userPoint = new UserPoint(mContext);
+
+        // testing if the constructor works correctly
+        ElevationMap elevationMap = new ElevationMap(userPoint);
+
+        Assert.assertTrue(true);
+
+    }
+/**
+ * Check that getTopography method does effectively return a non-null value
+ */
+    @Test
+    public void getTopographyMapBitmap() {
+
+        Context mContext = ApplicationProvider.getApplicationContext();
+
+        // setting everest location
+        UserPoint userPoint = new UserPoint(mContext);
+        userPoint.setLocation(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON,GPSTracker.DEFAULT_ALT, GPSTracker.DEFAULT_ACC);
+
+        ElevationMap elevationMap = new ElevationMap(userPoint);
+
+        // checking if there are no errors retrieving the AAIGrid
+        int[][] topographyMap = elevationMap.getTopographyMap();
+        Assert.assertFalse(topographyMap == null);
+
+    }
+/**
+ *  Check the changing of the location to get Topography Map
+ * Check with coordinates under the treshold and above
+ */
+    @Test
+    public void updateMapTest() {
```

Comment:
We have not really talked about how we want to do tests. So maybe it is just my personal opinion, we can discuss it on Discord ;) 
But I personally prefer if this is two tests. One for when we change the location to under the threshold and one where we change it above. That way if it is just one of the parts that has been broken only one test would fail and we would now what happened. 
But as I said maybe that is my personal opinion and not a must following convention.

Reply:
I think these tests should stay together as they test the same method (and the same if statement, with two different outcomes)